### PR TITLE
Replace "Site visibility" with a link to Calypso

### PR DESCRIPTION
**Pull request by okmttdhr**
Related to https://github.com/Automattic/dotcom-forge/issues/6694, https://github.com/Automattic/wpcomsh/pull/1893

## Proposed changes:

This PR replaces "Site visibility" with a link to Calypso on the Reading Settings page in wp-admin (i.e., This PR fixes an unexpected behavior: the Site Visibility setting is shown in wp-admin of Atomic Classic). 

### before
- Simple Classic
	- Screenshot 2024-05-31 at 13 54 38
- Atomic Classic
	- Screenshot 2024-05-31 at 13 55 55

### after
- Simple Classic, Atomic Classic 
	- Screenshot 2024-05-31 at 13 54 02
- Atomic Classic without Jetpack connected
	- private site
		- Screenshot 2024-05-31 at 15 36 44
	- public site
		- Screenshot 2024-05-31 at 15 55 14

---
**The Site Visibility setting should just be a link to Calypso**. See https://github.com/Automattic/dotcom-forge/issues/6526#issuecomment-2116939266 for more details. 

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Atomic Classic

- Prepare an Atomic Classic site with the flag `wpcom_classic_early_release=1`
- Patch this PR
- Go to Settings > Reading
	- See `Site visibility > Manage your site visibility settings`
- Click `Manage your site visibility settings` and set your site `Coming Soon` or `Private`
	- Go to Jetpack > My Jetpack
	- Disconnect Jetpack from the bottom of the page
	- Go to Settings > Reading
	- See `Site visibility > Jetpack is disconnected & site is private. Reconnect Jetpack to manage site visibility settings.`
- Go to Jetpack > My Jetpack
	- Reconnect Jetpack
	- Go to Settings > Reading
	- Click `Manage your site visibility settings` and set your site `Public`
	- Go to Jetpack > My Jetpack
	- Disconnect Jetpack from the bottom of the page
	- Go to Settings > Reading
	- See the setting as the same as Core
	- Confirm the option can be updated

Simple Classic

- Prepare an Atomic Classic site with the flag `wpcom_classic_early_release=1`
- Patch this PR
- Go to Settings > Reading
- See `Site visibility > Manage your site visibility settings`

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/replace-site-visibility/replace-site-visibility.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ */
+function is_jetpack_connected() {
+	// @phan-suppress-next-line PhanUndeclaredClassMethod, PhanUndeclaredClassInCallable
+	return class_exists( 'Jetpack' ) && is_callable( 'Jetpack::is_connection_ready' ) && Jetpack::is_connection_ready();
+}
+
+/**
+ * Replaces the 'Site Visibility' privacy options selector with a Calypso link.
+ */
+function replace_site_visibility() {
```
Comment (okmttdhr) [inline projects/packages/jetpack-mu-wpcom/src/features/replace-site-visibility/replace-site-visibility.php:21]: The code was mostly ported from https://github.com/Automattic/wpcomsh/pull/1893

**Review by miksansegundo — DISMISSED** [review]
LGTM 🚀 

Tested well on all scenarios:

- Simple and Atomic Classic
  - with and without Jetpack connected
  - with the coming soon, public, and private settings

There is an issue on Calypso's side because the copy of the link doesn't connect with the heading of the section. The heading says "Launch" or "Privacy" after the site is launched.

**Manage your site visibility settings -> Launch site**

Screenshot 2567-05-31 at 16 18 52

**Manage your site visibility settings -> Privacy**

Screenshot 2567-05-31 at 16 02 21

I suggest we update the headings and unify the copy to **Visibility settings** or **Site visibility**. cc: @lucasmendes-design @taipeicoder 

I'll go ahead and create an [issue for discussion](https://github.com/Automattic/dotcom-forge/issues/7547).

**Comment by okmttdhr** [comment]
Thank you for your testing and following-up, @miksansegundo! 

I'm merging this PR!

**Comment by okmttdhr** [comment]
@Automattic/lego @Automattic/explorers @miksansegundo Could someone approve this PR again, as I merged the trunk? Thank you!

**Review by taipeicoder — APPROVED** [review]

**Pull request merged**